src/config.rs

```diff
@@ -0,0 +1,100 @@
+//! Implementation of the configuration module.
+
+use anyhow::Context;
+use anyhow::Result;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::Mode;
+
+/// Represents the configuration for the Sprocket CLI tool.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct Config {
+    /// Configuration for the `format` command.
+    #[serde(rename = "format")]
+    pub format_config: FormatConfig,
+    /// Configuration for the `check` and `lint` commands.
+    #[serde(rename = "check")]
+    pub check_config: CheckConfig,
+    /// Configuration for the `validate` command.
+    #[serde(rename = "validate")]
+    pub validate_config: ValidateInputsConfig,
+}
+
+/// Represents the configuration for the Sprocket `format` command.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct FormatConfig {
+    /// Disables color output.
+    pub no_color: bool,
+    /// The report mode.
+    pub report_mode: Option<Mode>,
+    /// Use tabs for indentation (default is spaces).
+    pub with_tabs: bool,
+    /// The number of spaces to use for indentation levels (default is 4).
+    pub indentation_size: Option<usize>,
+    /// The maximum line length (default is 90).
+    pub max_line_length: Option<usize>,
+    /// Overwrite the WDL documents with the formatted versions.
+    pub overwrite: bool,
+    /// Check if the files are formatted correctly and print diff if not.
+    pub check: bool,
+}
+
+/// Represents the configuration for the Sprocket `check` and `lint` commands.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct CheckConfig {
+    /// Rule IDs to except from running.
+    pub except: Vec<String>,
+    /// Causes the command to fail if any warnings are reported.
+    pub deny_warnings: bool,
+    /// Causes the command to fail if any notes are reported.
+    pub deny_notes: bool,
+    /// Suppress diagnostics from imported documents.
+    pub single_document: bool,
+    /// Show diagnostics for remote documents.
+    pub show_remote_diagnostics: bool,
+    /// Run the `shellcheck` program on command sections.
+    pub shellcheck: bool,
+    /// Hide diagnostics with `note` severity.
+    pub hide_notes: bool,
+    /// Disables color output.
+    pub no_color: bool,
+    /// The report mode.
+    pub report_mode: Option<Mode>,
+}
+
+/// Represents the configuration for the Sprocket `validate` command.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct ValidateInputsConfig {
```

Comment:
Said this over slack, but I think `no_color` and `report_mode` should probably be moved to the root of the config, and then we can drop the `ValidateInputsConfig` entirely